public/app/plugins/panel/text/module.tsx

```diff
@@ -27,6 +27,15 @@ export const plugin = new PanelPlugin<PanelOptions>(TextPanel)
         description: 'Content of the panel',
         editor: TextPanelEditor,
         defaultValue: defaultPanelOptions.content,
+      })
+      .addBooleanSwitch({
+        path: 'query',
+        name: 'Execute query',
+        description: 'Run a query and include the results in template variable context',
+        defaultValue: defaultPanelOptions.query,
       });
   })
+  .setSkipDataQuery((opts?: PanelOptions) => {
+    return opts?.query !== true;
+  })
```

Comment:
Thinking out loud: I would consider `opts.query` to be a common option for certain panels. IMO `setSkipDataQuery` be just a plain call from module definition POV (`.setSkipDataQuery()`) where implementation of the function responsible for the check would be provided by `PanelPlugin`. 

The implementation here assumes the query being/not being skipped may depend on other options - not sure if this isn't a little bit too much of a flexibility.

Reply:
The point of this is have *something* dynamic define if it should have a query section or not.  PanelPlugin is a singleton instance for all panels of that type -- we need some way to ask if the specific instance should have a query or not.